reactive/kubernetes_worker.py

```diff
@@ -1073,25 +1083,30 @@ def restart_unit_services():
         service_restart('snap.%s.daemon' % service)
 
 
-def get_kube_api_servers(kube_api):
-    '''Return the kubernetes api server address and port for this
-    relationship.'''
-    hosts = []
-    # Iterate over every service from the relation object.
-    for service in kube_api.services():
-        for unit in service['hosts']:
-            hosts.append('https://{0}:{1}'.format(unit['hostname'],
-                                                  unit['port']))
-    return hosts
+def get_kube_api_servers():
+    '''Return the list of kubernetes API endpoint URLs.'''
+    kube_control = endpoint_from_name("kube-control")
+    kube_api = endpoint_from_name("kube-api-endpoint")
+    # prefer kube-api-endpoints
+    if kube_api.services():
+        return [
+            'https://{0}:{1}'.format(unit['hostname'], unit['port'])
+            for service in kube_api.services()
+            for unit in service['hosts']
+        ]
+    if hasattr(kube_control, "get_api_endpoints"):
+        return kube_control.get_api_endpoints()
+    return []
```

Comment:
This is one of those "should never happen" cases, and if encountered, will lead to hook errors since callers assume that at least 1 valid endpoint is returned. I would think it would be more fitting to raise an exception here, or at the very least, log a warning.